From 2468fb596a80da386e3fc59a7a4f52cf4d8748dc Mon Sep 17 00:00:00 2001
From: Lukasz Bartosik <lbartosik@marvell.com>
Date: Fri, 4 Jan 2019 16:36:17 +0100
Subject: [PATCH 0900/1051] crypto: cpt - fix invalid IV size for ecb mode

Change-Id: Ia20315e257dcef62980d6a79e42a4401db6dd0a0
Signed-off-by: Lukasz Bartosik <lbartosik@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/2676
Tested-by: sa_ip-sw-jenkins
Reviewed-by: Srujana Challa <schalla@cavium.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/crypto/cavium/cpt/common/cpt_algs.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/crypto/cavium/cpt/common/cpt_algs.c b/drivers/crypto/cavium/cpt/common/cpt_algs.c
index 3272b518e264..02025a762424 100644
--- a/drivers/crypto/cavium/cpt/common/cpt_algs.c
+++ b/drivers/crypto/cavium/cpt/common/cpt_algs.c
@@ -1216,7 +1216,7 @@ struct crypto_alg algs[] = { {
 	.cra_type = &crypto_ablkcipher_type,
 	.cra_u = {
 		.ablkcipher = {
-			.ivsize = AES_BLOCK_SIZE,
+			.ivsize = 0,
 			.min_keysize = AES_MIN_KEY_SIZE,
 			.max_keysize = AES_MAX_KEY_SIZE,
 			.setkey = cvm_ecb_aes_setkey,
@@ -1281,7 +1281,7 @@ struct crypto_alg algs[] = { {
 		.ablkcipher = {
 			.min_keysize = DES3_EDE_KEY_SIZE,
 			.max_keysize = DES3_EDE_KEY_SIZE,
-			.ivsize = DES_BLOCK_SIZE,
+			.ivsize = 0,
 			.setkey = cvm_ecb_des3_setkey,
 			.encrypt = cvm_encrypt,
 			.decrypt = cvm_decrypt,
-- 
2.17.1

